lib/client.js

```diff
@@ -83,6 +83,46 @@ function isValidEvent(client, event) {
   return client && client._origin === event.origin && client._source === event.source;
 }
 
+function triggerEvent(client, eventName, data) {
+  if (!client._messageHandlers[eventName]) { return false; }
+  client._messageHandlers[eventName].forEach(function(handler) {
+    handler(data);
+  });
+}
+
+function finalizePendingPromise(pendingPromise, data) {
+  if (data.error) {
+    var error = data.error.name ? createError(data.error) : data.error;
+    pendingPromise.reject(error);
+  } else {
+    pendingPromise.resolve(data.result);
+  }
+}
+
+function postReplyWith(client, key, msg) {
+  if (client._repliesPending[key]) { return; }
+  msg.key = 'iframe.reply:' + key;
+  client._repliesPending[key] = true;
+  when(client._messageHandlers[key]).then(
+    rawPostMessage.bind(null, client, msg)
+  ).catch(function(reason) {
+    msg.error = {
+      msg: reason
+    };
+    rawPostMessage(client, msg);
+  }).then(function() {
+    delete client._repliesPending[key];
+  });
+}
+
+function getMessageRecipient(client, recipientGuid) {
+  if (recipientGuid && recipientGuid !== client._instanceGuid) {
+    return client._instanceClients[recipientGuid];
```

Comment:
Mmm I don't see how, because if the framework knows posts a message to an iframe with an `instanceGuid` it's because this iframe create a client for that instance at some point, and we're never clearing `_instanceClients` right now. But I agree we should still handle that. I'll make it log an error.